applications/MeshingApplication/python_scripts/mmg_process.py

```diff
@@ -417,12 +419,38 @@ def ExecuteInitializeSolutionStep(self):
                                 self.step = 0  # Reset
                                 self.time = 0.0  # Reset
 
+    def ExecuteFinalize(self):
+        """ This method is executed in order to finalize the simulation and save the refined mesh in a new .mdpa file
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        self.average_remeshing = self.settings["remesh_post_process"].GetBool()
+        output_file_path = self.settings["output_file_name"].GetString()
+        if self.average_remeshing:
+            self._ExecuteRefinement()
+            if self.main_model_part.HasSubModelPart("fluid_computational_model_part"):
+                self.main_model_part.RemoveSubModelPart("fluid_computational_model_part")
+            KratosMultiphysics.ModelPartIO(output_file_path, KratosMultiphysics.IO.WRITE | KratosMultiphysics.IO.MESH_ONLY).WriteModelPart(self.main_model_part)
```

Comment:
Here we lack another boolean, to control if the mesh is going to be written or not. Take into account that we might want to remesh the mesh at the final step but no necessarily write it to a file.

```suggestion
            if self.output_final_mesh
                KratosMultiphysics.ModelPartIO(output_file_path, KratosMultiphysics.IO.WRITE | KratosMultiphysics.IO.MESH_ONLY).WriteModelPart(self.main_model_part)
```

Also to be read from the settings.